### PR TITLE
NodeJS SDK - Handle malformed responses

### PR DESCRIPTION
**Pull request by richardscarrott**
When userGrid is down or returns something unexpected this response handler throws an exception because it tries to access properties of undefined.

### CODE REVIEW

**Comment by richardscarrott** [comment]
There's possibly other areas that need proper checks / try catch's but I just wanted to highlight the issue.

File: sdks/nodejs/lib/usergrid.js [hunk 1]
```diff
@@ -145,6 +147,7 @@ var AUTH_NONE = 'NONE';
             callback(err, data);
           }
         } else {
+          r = r.body || {};
```
Comment (shawnfeldman) [inline sdks/nodejs/lib/usergrid.js]: do you mean r = r || {} ? or does r.body contain another body?

**Comment by richardscarrott** [comment]
Yeah my bad, it should be `r.body = r.body || {}` -- tbh the code was just edited in Github's editor to explain the issue, I'm sure there's a more elegant approach and a number of other places that need to be checked, e.g. the data that's passed into the callback isn't guaranteed to be an object.

**Comment by richardscarrott** [comment]
I might be able to send a proper pull request tomorrow if it helps?

**Comment by shawnfeldman** [comment]
if you fix it i'll merge the pr, also i'll probably run "node test.js" to verify the change works.  https://github.com/usergrid/usergrid/tree/master/sdks/nodejs#about-the-samples

**Comment by richardscarrott** [comment]
I took a better look at this today and have updated the pull request -- I think the only assumptions that shouldn't be made in the callback is the existence of body.

Also, the test suite still passes.

**Pull request merged**